### PR TITLE
[CI][OPENCL] adapt e2e workflow to opencl target

### PR DESCRIPTION
**Pull request by wlemkows**
- [x] Waits for: https://github.com/oneapi-src/unified-runtime/pull/993 ([source/adapters/opencl/CMakeLists.txt](https://github.com/oneapi-src/unified-runtime/pull/993/files#diff-097e8a00604cf0f1635f4a7c6917bc25e00ad1375221c1a39b115934c91ccfb6))

- [x] build pass: -https://github.com/oneapi-src/unified-runtime/actions/runs/7221868465/job/19677743227

### CODE REVIEW

File: .github/workflows/e2e_nightly.yml [hunk 1]
```diff
@@ -111,7 +119,7 @@ jobs:
         -GNinja
         -B ${{github.workspace}}/build-e2e/
         -S ${{github.workspace}}/sycl-repo/sycl/test-e2e/
-        -DSYCL_TEST_E2E_TARGETS="ext_oneapi_cuda:gpu"
+        -DSYCL_TEST_E2E_TARGETS="${{matrix.adapter.prefix}}_${{matrix.adapter.str_name}}:${{matrix.adapter.unit}}"
```
Comment (lukaszstolarczuk) [inline .github/workflows/e2e_nightly.yml:122]: this `_` between prefix and name seems not right 😉

File: .github/workflows/e2e_nightly.yml [hunk 2]
```diff
@@ -1,17 +1,19 @@
...
+#  schedule:
+#    # Run every day at 23:00 UTC
+#    - cron: '0 23 * * *'
 
 jobs:
   e2e-build-hw:
     name: Build SYCL, UR, run E2E
     strategy:
       matrix:
         adapter: [
```
Comment (lukaszstolarczuk) [inline .github/workflows/e2e_nightly.yml:14]: you could add `fail-fast: false` - we may want to run all jobs independently (at least, as long as they run in nightly job)

**Review by lukaszstolarczuk — COMMENTED** [review]
in general it looks good 👍

File: .github/workflows/e2e_nightly.yml (hunk 1 above)
Comment (wlemkows) [reply .github/workflows/e2e_nightly.yml:122]: This is intentional. It is part of the target name.

Comment (lukaszstolarczuk) [reply .github/workflows/e2e_nightly.yml:122]: What I meant is - it won't work in case of an empty prefix...?

Comment (wlemkows) [reply .github/workflows/e2e_nightly.yml]: Fixed.

File: .github/workflows/e2e_nightly.yml (hunk 2 above)
Comment (wlemkows) [reply .github/workflows/e2e_nightly.yml:15]: I think it should fail to not block other runners if anything is failing.

**Comment by wlemkows** [comment]
Runner set, PR passes, ready to merge.

**Review by pbalcer — APPROVED** [review]

**Review by lukaszstolarczuk — APPROVED** [review]

**Pull request merged**